### PR TITLE
Refactored code and Added cni addon upgrade/downgrade regression test

### PR DESCRIPTION
**Pull request by cgchinmay**
Addon upgrade/downgrade test similar to https://github.com/aws/amazon-vpc-cni-k8s/pull/1795

Added tests for addon upgrade/downgrade

**What type of PR is this?**
feature
testing
cleanup

**Which issue does this PR fix**:
Adds integration tests for testing upgrade/downgrade using user provided initial and final CNI version

**What does this PR do / Why do we need it**:
This is a regression test after cni addon upgrade/downgrade

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:
```
ginkgo -v -r -- --cluster-kubeconfig=/Users/cgadgil/.kube/config --cluster-name=cni-rc-test --aws-region=us-west-2 --aws-vpc-id=vpc-0faed7c6e93018b48 --ng-name-label-key=eks.amazonaws.com/nodegroup --ng-name-label-val=nodegroup
Running Suite: CNI Pod Networking Suite
=======================================
Random Seed: 1644821135
Will run 2 of 2 specs

STEP: creating test namespace
STEP: getting the node with the node label key eks.amazonaws.com/nodegroup and value nodegroup
STEP: verifying that atleast 1 nodes is present for the test
STEP: getting the instance type from node label beta.kubernetes.io/instance-type
STEP: getting the network interface details from ec2
STEP: describing the VPC to get the VPC CIDRs
STEP: setting the environment variables on the ds to map[WARM_ENI_TARGET:0 WARM_IP_TARGET:3]
STEP: getting the aws-node daemon set in namesapce kube-system
STEP: updating the daemon set with new environment variable
cni addon upgrade/downgrade test test host networking 
  create test deployment with initial addon and verify deletion with target addon
  /Users/cgadgil/Documents/amazon-vpc-cni-k8s/test/integration-new/regression-tests/cni_addon_test.go:31
STEP: using target addon version as v1.10.1-eksbuild.1
STEP: using initial addon version as v1.7.5-eksbuild.2
STEP: delete existing cni addon if exists
STEP: wait till the old addon is deleted
STEP: install the new addon with version: v1.7.5-eksbuild.2
STEP: wait till the addon is active
...
...
...
cni addon upgrade/downgrade test
/Users/cgadgil/Documents/amazon-vpc-cni-k8s/test/integration-new/regression-tests/cni_addon_test.go:29
  test host networking
  /Users/cgadgil/Documents/amazon-vpc-cni-k8s/test/integration-new/regression-tests/cni_addon_test.go:30
    create test deployment with target addon and verify deletion with initial addon
    /Users/cgadgil/Documents/amazon-vpc-cni-k8s/test/integration-new/regression-tests/cni_addon_test.go:63
------------------------------
STEP: deleting test namespace
STEP: update environment variables map[], remove map[WARM_ENI_TARGET:{} WARM_IP_TARGET:{}]
STEP: getting the aws-node daemon set in namesapce kube-system
STEP: updating the daemon set with new environment variable

Ran 2 of 2 Specs in 579.694 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/integration-new/regression-tests/cni_addon_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+var (
+	err         error
+	podLabelKey = "app"
+	podLabelVal = "host-networking-test"
+	deployment  *v1.Deployment
+)
+
+const (
+	DEFAULT_VERSION = "v1.7.5-eksbuild.2"
+	LATEST_VERSION  = "v1.10.1-eksbuild.1"
```
Comment (jayanthvn) [inline test/integration-new/regression-tests/cni_addon_test.go:26]: LATEST_VERSION can be set to default latest release which will be useful for daily or weekly runs. But when user runs this test, we need to provide an option for them to pass the built image (with their fix) or even better if we can build the image and pass it to this test if possible.

File: test/framework/resources/k8s/utils/addon.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+func WaitTillAddonIsActive(eks services.EKS, addonName string, clusterName string) error {
+	ctx := context.Background()
+	return wait.PollImmediateUntil(utils.PollIntervalShort, func() (bool, error) {
+		describeAddonOutput, err := eks.DescribeAddon(&services.AddonInput{
+			AddonName:   addonName,
+			ClusterName: clusterName,
+		})
+		status := *describeAddonOutput.Addon.Status
+		if status == "ACTIVE" {
```
[the remark this answers is not in the document]
Comment (cgchinmay) [reply test/framework/resources/k8s/utils/addon.go:33]: Yes, will add that

File: test/integration-new/regression-tests/cni_addon_test.go (hunk 1 above)
Comment (cgchinmay) [reply test/integration-new/regression-tests/cni_addon_test.go:26]: Yes, there is an option to specify initial and target addons. If those are not provided then we fallback to these values.
We do want to be able to provide a custom built image url so I am working on a bash script which will invoke the ginkgo tests. Similar to 'run-canary-test.sh' that we have right now. 
Applying manifest files via ginkgo requires you to create every single resource in manifest, don't think there is readily available equivalent for 'kubectl apply'.So am working on bash script, while this is for published addons and can be treated like a regression test.

Comment (jayanthvn) [reply test/integration-new/regression-tests/cni_addon_test.go:26]: Sounds good. Thanks.

File: test/framework/resources/k8s/utils/addon.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cgchinmay) [reply test/framework/resources/k8s/utils/addon.go:41]: CrashLoopBack is context of pod. Addons don't have such status. So I have added check in test instead to ensure that the pods are actually running.

[the remark this answers is not in the document]
Comment (jayanthvn) [reply test/framework/resources/k8s/utils/addon.go:41]: sounds good.

**Review by jayanthvn — DISMISSED** [review]

**Review by jayanthvn — APPROVED** [review]

**Pull request merged**